app/routes/__index/domains/new.tsx

```diff
@@ -1,11 +1,66 @@
-import { Heading } from '@chakra-ui/react';
+import { Container, Heading, Text } from '@chakra-ui/react';
+import { json, redirect } from '@remix-run/node';
+import { RecordType } from '@prisma/client';
+import { useActionData } from '@remix-run/react';
+import { z } from 'zod';
+import { parseFormSafe } from 'zodix';
+
+import DnsRecordForm from '~/components/dns-record/form';
+import { createRecord } from '~/models/record.server';
+import { requireUsername } from '~/session.server';
+
+import type { ActionArgs } from '@remix-run/node';
+import type { ZodError } from 'zod';
+
+function errorForField(error: ZodError, field: string) {
+  return error.issues.find((issue) => issue.path[0] === field)?.message;
+}
+
+export const action = async ({ request }: ActionArgs) => {
+  // Create a Zod schema for validation
+  // Optional is not needed as we get '' if nothing is entered
+  const DnsRecord = z.object({
+    name: z.string().min(1), // We do not want to consider '' a valid string
+    type: z.nativeEnum(RecordType),
+    value: z.string().min(1),
+    ports: z.string(),
+    course: z.string(),
+    description: z.string(),
+  });
+
+  const newDnsRecordParams = await parseFormSafe(request, DnsRecord);
+  const username = await requireUsername(request);
+
+  // If validations failed, we return the errors to show on the form
+  // Currently only returns 'type' field errors as no other validations exist
+  // Also, form cannot be submitted without required values
+  if (newDnsRecordParams.success === false) {
+    return json({
```

Comment:
Do we need to use `typedjson` here?  cc @Myrfion

Reply:
As far as I understand no dates or other types that might require `typedjson` are returned in response, so I don't see a necessity for it there, just only if you think we should use it everyevere

Reply:
I'd suggest using the same approach throughout the code.